receiver/vcenterreceiver/client.go

```diff
@@ -0,0 +1,209 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package vcenterreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/vcenterreceiver"
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+	"time"
+
+	"github.com/vmware/govmomi"
+	"github.com/vmware/govmomi/find"
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/govmomi/performance"
+	"github.com/vmware/govmomi/property"
+	"github.com/vmware/govmomi/vim25"
+	vt "github.com/vmware/govmomi/vim25/types"
+	"github.com/vmware/govmomi/vsan"
+	"github.com/vmware/govmomi/vsan/types"
+)
+
+// vcenterClient is a client that
+type vcenterClient struct {
+	moClient   *govmomi.Client
+	vimDriver  *vim25.Client
+	vsanDriver *vsan.Client
```

Comment:
When working with a `vcenterClient`, do I have to worry about the state of the internal clients? If so, I'm not sure we want to wrap them up like this.

Reply:
From my current knowledge and looking at the govmomi codebase, it appears that it will handle most reconnections and auth changes for us. Is there a particular case you want me to test?

Reply:
Just trying to figure out if it's really necessary to expose a `ConnectVSAN` function. Based on what you're saying it seems like this is an internal concern for the client.

Reply:
I guess the tricky part is switching the control flow for this, on the scraper I have introduced a `vsanEnabled` toggle that determines when to do the vSAN data retrieval and processing for portions for different resources. Potentially another misnomer or a better name we could use?